backend/app/rag/node_postprocessor/vllm_reranker.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, List, Optional
+import requests
+
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.callbacks import CBEventType, EventPayload
+from llama_index.core.instrumentation import get_dispatcher
+from llama_index.core.instrumentation.events.rerank import (
+    ReRankEndEvent,
+    ReRankStartEvent,
+)
+from llama_index.core.postprocessor.types import BaseNodePostprocessor
+from llama_index.core.schema import MetadataMode, NodeWithScore, QueryBundle
+
+dispatcher = get_dispatcher(__name__)
+
+
+class VLLMRerank(BaseNodePostprocessor):
+    api_url: str = Field(default="", description="API url.")
+    model: str = Field(default="", description="The model to use when calling API.")
+
+    top_n: int = Field(description="Top N nodes to return.")
+
+    _session: Any = PrivateAttr()
+
+    def __init__(
+        self,
+        top_n: int = 2,
+        model: str = "BAAI/bge-reranker-v2-m3",
+        api_url: str = "http://localhost:8000/v1/score",
```

Comment:
Recommended to refer to Cohere's and Xinference's implementations:

```suggestion
        base_url: str = "http://localhost:8000",
```

Cohere:

https://github.com/run-llama/llama_index/blob/07a832007796cb5fea600c4ca00e04205b0ec281/llama-index-integrations/postprocessor/llama-index-postprocessor-cohere-rerank/llama_index/postprocessor/cohere_rerank/base.py#L20C5-L20C13

XinferenceRerank:

https://github.com/run-llama/llama_index/blob/07a832007796cb5fea600c4ca00e04205b0ec281/llama-index-integrations/postprocessor/llama-index-postprocessor-xinference-rerank/llama_index/postprocessor/xinference_rerank/base.py#L27